docs/articles/vector_search_based_recommender_system.md

```diff
@@ -0,0 +1,276 @@
+# Build a Vector-Search based Recommender System
+
+## Overview
+
+In this guide, you will:
+- Learn how to use a reliable and performant database optimized for vectors (Rockset) along with OpenAI embeddings to build a recommendation engine
+- Build a dynamic web application using vanilla CSS, HTML, JavaScript, and Flask, that seamlessly integrates with Rockset and OpenAI APIs to create a recommendation system.
+- Find an end-to-end Colab notebook that you can run without any dependencies on your local operating system: [Recsys_workshop](https://colab.research.google.com/drive/1rD08fTiCGJiKFDDNZyJh92QN4_dz6Hcu)
+
+## Introduction
+A real-time recommender system with an optimized and efficient [architecture](https://rockset.com/blog/a-blueprint-for-a-real-world-recommendation-system/) can add value to an organization by enhancing personalization, user engagement, and ultimately increasing user satisfaction.
+
+Building a recommendation system that efficiently deals with high-dimensional data to find accurate, relevant, and similar items in a large dataset requires effective and efficient vectorization, vector indexing, search, and retrieval. This, in turn, demands robust databases with vector capabilities. For this post, we will use Rockset as the database and OpenAI embedding models to vectorize the dataset.  Other vector databases are available, you can find examples [here](https://vdbs.superlinked.com/).
+
+There are several components of vector search required to build a recommendation engine, we won't dive into them here however you can read more about them [here](https://superlinked.com/vectorhub/building-blocks/vector-search/introduction). The key pieces for this example are a database optimized for metadata filtering, vector search, and keyword search, supporting sub-second search, aggregations, and joins at scale, is needed.
+
+**Overview of the Recommendation WebApp**
+The image below shows the workflow of the application we'll be building. We have unstructured data i.e., game reviews in our case. We'll generate vector embeddings for all of these reviews through OpenAI model and store them in the database. Then we'll use the same OpenAI model to generate vector embeddings for our search query and match it with the review vector embeddings using a similarity function such as the nearest neighbor search, dot product or approximate neighbor search. Finally, we will have our top 10 recommendations ready to be displayed.
```

Comment:
I think the last part is a bit ambiguous. Dot product is a similarity function, while (approximate) nearest neighbor searches are algorithms (which can btw utilise dot product - or other, like cosine- similarity)

Reply:
good catch, let's change it to, "using a similarity measure such as approximate neighbor search"